web/app/editor/ExpandablePropertiesEditor.jsx

```diff
@@ -1,30 +1,51 @@
 import React from 'react'
 import {modelLookup} from './EditorModel.js'
 import {PropertiesEditor, shouldShowProperty} from './PropertiesEditor'
-import {modelProperties, wrapOptional} from './EditorModel'
-import {modelProperty} from './EditorModel'
-import {navigateWithQueryParams, currentLocation} from '../util/location'
+import {modelErrorMessages, modelProperties, modelProperty, wrapOptional} from './EditorModel'
+import {currentLocation, navigateWithQueryParams} from '../util/location'
 import {parseBool} from '../util/util'
 import Text from '../i18n/Text'
+import {isEmpty} from 'ramda'
+import * as Bacon from 'baconjs'
 
-export const ExpandablePropertiesEditor = ({model, propertyName, propertyFilter = () => true}) => {
-  let propertyModel = modelLookup(model, propertyName)
-  let edit = model.context.edit
-  let paramName = propertyName + '-expanded'
-  let expanded = parseBool(currentLocation().params[paramName])
-  let wrappedModel = edit ? wrapOptional(propertyModel) : propertyModel
-  let toggleOpen = () => {
-    navigateWithQueryParams({[paramName]: !expanded ? 'true' : undefined})
+export class ExpandablePropertiesEditor extends React.Component {
```

Comment:
Would it be easier (after some time passes) to reason about this component if we kept it stateless, and instead used a wrapper component to decorate with the new functionality (= causing the expansion side effect)?